organization/views.py

```diff
@@ -226,6 +229,9 @@ def get_form_kwargs(self):
         form_kwargs['organization_id'] = self.kwargs['slug']
         return form_kwargs
 
+    def get_cancel_url(self):
```

Comment:
I put this additional implementation of `get_cancel_url` in because I couldn't get the the URL to work with a pattern that didn't use `slug` or `pk`.

Reply:
The trick here is that we need to update the URLconf for this view to use the name `organization_id` for the slug:

```diff
--- a/organization/urls.py
+++ b/organization/urls.py
@@ -20,7 +20,7 @@ urlpatterns = [
     url(r'name/autocomplete',
         organization_autocomplete,
         name="organization-autocomplete"),
-    url(r'edit/(?P<slug>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4
}-[0-9a-f]{12})/$',
+    url(r'edit/(?P<organization_id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}
-[0-9a-f]{4}-[0-9a-f]{12})/$',
         OrganizationEditBasicsView.as_view(),
         name='edit-organization'),
```

Then, we need to set the [`slug_url_kwarg` attribute](https://docs.djangoproject.com/en/2.2/ref/class-based-views/mixins-single-object/#django.views.generic.detail.SingleObjectMixin.slug_url_kwarg) on the `OrganizationEditView` class to tell Django that `organization_id` is the model slug (we also have an attribute named `slug_field_kwarg`, which I don't understand, because that's not a Django attribute... but for backwards-compatibility we should probably just keep supporting it):

```diff
--- a/organization/views.py
+++ b/organization/views.py
@@ -169,7 +169,8 @@ class OrganizationDetail(BaseDetailView):
 class OrganizationEditView(EditButtonsMixin, BaseUpdateView):
     model = Organization
     slug_field = 'uuid'
-    slug_field_kwarg = 'slug'
+    slug_field_kwarg = 'organization_id'
+    slug_url_kwarg = 'organization_id'
     context_object_name = 'organization'
     button = 'basics'
```

We'll probably also want to update `OrganizationEditBasicsView.get_reference_organization` to use `organization_id` instead of `slug`, and remove `OrganizationEditBasicsView.get_form_kwargs`, too:

```diff
@@ -210,7 +211,7 @@ class OrganizationEditBasicsView(OrganizationEditVi
ew):
     button = 'basics'
 
     def get_reference_organization(self):
-        return Organization.objects.get(uuid=self.kwargs['slug'])
+        return Organization.objects.get(uuid=self.kwargs['organization
_id'])
 
     def get_context_data(self, **kwargs):
+        return Organization.objects.get(uuid=self.kwargs['organization
_id'])
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
@@ -224,13 +225,10 @@ class OrganizationEditBasicsView(OrganizationEdit
View):
         else:
             return super().get_success_url()
 
-    def get_form_kwargs(self):
-        form_kwargs = super().get_form_kwargs()
-        form_kwargs['organization_id'] = self.kwargs['slug']
-        return form_kwargs
```

Once all that's done, you should be able to remove `get_cancel_url` here and inherit the version on the base class instead!